File: sdk/javascript/test_vines.js
const fs = require('fs');
const pathActual = require('path');
// Replace './ScyKernel' with the actual path to your class file
const ScyKernel = require('./ScyKernel'); 

async function runTest() {
    const dir = "vines_images";
    const path = pathActual.join(dir, "node_vine.ppm");
    const path2 = pathActual.join(dir, "node_vine.png");
    const testKey = "User";
    const testValue = "Amanda";
    const password = "ScyWeb_Global_Secret_2026";

    // Ensure the local folder exists
    if (!fs.existsSync(dir)) {
        fs.mkdirSync(dir);
    }
    
    // Instantiating 'scy' with the password and the local path
    const scy = new ScyKernel(password, path);

    // Creating the test DBs
    scy._createPPM_DB(path);
    scy._syncPNG(path2, "load");

    // Test both PPM and PNG DBs
    scy._putToPPM(testKey, testValue, password);
    scy._putToPNG(testKey, testValue, password);

    // sync changes and refresh
    scy._syncPNG(path2, "commit");
    scy._syncPNG(path2, "load");

    // Retrieve the results from both DBs
    const result = scy._getFromPPM(testKey, password);
    const result2 = scy._getFromPNG(testKey, password);

    // Output Comparison
    if (result === testValue && result2 === testValue) {
        const isValid = scy._validateDB(path) ? "Valid" : "Invalid";
        console.log(`✅ Node.js KV Parity: SUCCESS (Recovered: ${result})`);
        console.log(`🧩 PPM is: ${isValid}`);
        const size = scy._getFileSize(path2);
        console.log(`📏 Size of Image DB: ${size} bytes`);
        // fs.unlinkSync(path);
        // fs.unlinkSync(path2);
        const parityConfigs = [
            ["C++", "../cpp/vines_images/cpp_vine.png"],
            ["Go", "../go/scykernel/vines_images/go_vine.png"],
            ["Java", "../java/vines_images/java_vine.png"],
            ["Node", "../javascript/vines_images/node_vine.png"],
            ["Kotlin", "../kotlin/vines_images/kt_vine.png"],
            ["PHP", "../php/vines_images/php_vine.png"],
            ["Python", "../python/vines_images/py_vine.png"],
            ["React Native", "../reace-native/vines_images/rn_vine.png"],
            ["Rust", "../rust/vines_images/rust_vine.png"],
            ["Swift", "../swift/vines_images/swift_vine.png"]
        ];
        for (const [lang, lPath] of parityConfigs) {
            if (fs.existsSync(lPath)) {
                const scyCheck = new ScyKernel(password, lPath);
                if (scyCheck._syncPNG(lPath, "load")) {
                    const res = scyCheck._getFromPNG(testKey, password);
                    if (res === testValue) {
                        console.log(`✅ JS to ${lang} Parity: SUCCESS (Recovered: ${res})`);
                    } else {
                        console.log(`❌ JS to ${lang} Parity: FAIL`);
                    }
                }
            }
        }
        process.exit(0);
    } else {
        console.error("❌ Node.js KV Parity: FAIL");
        console.error(`Expected: ${testValue}, Got PPM: [${result}]`);
        console.error(`Expected: ${testValue}, Got PNG: [${result2}]`);
        fs.unlinkSync(path);
        fs.unlinkSync(path2);
        process.exit(1);
    }
}

runTest();

File: sdk/rust/src/test_vines.rs
use scy_rust::ScyKernel;
use std::fs;
use std::path::Path;

fn main() {
    let dir = "vines_images";
    let path_ppm = format!("{}/rust_vine.ppm", dir);
    let path_png = format!("{}/rust_vine.png", dir);
    let test_key = "User";
    let test_value = "Amanda";
    let password = "ScyWeb_Global_Secret_2026";

    if !Path::new(dir).exists() {
        fs::create_dir_all(dir).expect("❌ Failed to create directory");
    }

    let mut scy = ScyKernel::new(password.to_string(), path_ppm.clone());

    scy.create_ppm_db(&path_ppm);
    scy.sync_png(&path_png, "load");

    scy.put_to_ppm(test_key, test_value, password);
    scy.put_to_png(test_key, test_value, password);

    scy.sync_png(&path_png, "commit");
    scy.sync_png(&path_png, "load");

    let result = scy.get_from_ppm(test_key, password);
    let result2 = scy.get_from_png(test_key, password);

    if result == test_value && result2 == test_value {
        let validation_test = if scy.validate_db(&path_ppm) { "Valid" } else { "Invalid" };
        println!("✅ Rust KV Parity: SUCCESS (Recovered: {})", result);
        println!("🧩 PPM is: {}", validation_test);
        
        let size = scy.get_file_size(&path_png);
        println!("📏 Size of Image DB: {} bytes", size);

        let parity_checks = vec![
            ("C++", "../cpp/vines_images/cpp_vine.png"),
            ("Go", "../go/scykernel/vines_images/go_vine.png"),
            ("Java", "../java/vines_images/java_vine.png"),
            ("Node", "../javascript/vines_images/node_vine.png"),
            ("Kotlin", "../kotlin/vines_images/kt_vine.png"),
            ("PHP", "../php/vines_images/php_vine.png"),
            ("Python", "../python/vines_images/py_vine.png"),
            ("React Native", "../react-native/vines_images/rn_vine.png"),
            ("Rust", "../rust/vines_images/rust_vine.png"),
            ("Swift", "../swift/vines_images/swift_vine.png")
        ];

        for (lang, p) in parity_checks {
            if Path::new(p).exists() {
                let mut scy_p = ScyKernel::new(password.to_string(), p.to_string());
                if scy_p.sync_png(p, "load") {
                    let res_p = scy_p.get_from_png(test_key, password);
                    if res_p == test_value {
                        println!("✅ Rust to {} Parity: SUCCESS (Recovered: {})", lang, res_p);
                    } else {
                        println!("❌ Rust to {} Parity: FAIL", lang);
                    }
                }
            }
        }
        std::process::exit(0);
    } else {
        println!("❌ Rust KV Parity: FAIL");
        println!("Expected: {}, Got: PPM [{}], PNG [{}]", test_value, result, result2);
        scy.delete_db(&path_ppm);
        scy.delete_db(&path_png);
        std::process::exit(1);
    }
}